packages/edit-site/src/components/global-styles/ui.js

```diff
@@ -276,120 +277,158 @@ function GlobalStylesEditorCanvasContainerLink() {
 	}, [ editorCanvasContainerView, isRevisionsOpen, goTo ] );
 }
 
-function GlobalStylesUI() {
+function NavigationSync( { path: parentPath, onPathChange, children } ) {
```

Comment:
It's unclear why is this a component. It should be just a hook.

Reply:
The reason why it is a component is to allow conditional application of the effect. A component can be applied conditionally while a hook can not.
```
const WrapperComponent = path && onPathChange ? NavigationSync : Fragment;
```

So for the uncontrolled case, we don't execute the effect at all. I think in this case a Component is a better choice, we can extract the effect as a hook anyway, having a component and a hook inside.

Reply:
I'm not sure, I still prefer a hook personally and just return early in the effect. I agree though that it's unfortunate that we can't use conditional hooks. I think component make more sense when we want to render something and not just return "children".

Reply:
Hi @youknowriad I tried to use the hook without a wrapper component but I noticed we need a wrapper component anyway because the hook calls need to be nested inside NavigatorProvider. So given that we are forced to have a wrapper I think the current solution is the preferred one.